operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,263 @@
+    std::copy_n(color.begin(), color.size(), start);
+  }
+
+  for (int64_t i = 1; i < thickness; ++i) {
+    auto xy_point_start = point_start + i * stride;
+    auto end = xy_point_start + line_length * image.channels;
+    std::copy_n(point_start, line_length * image.channels, xy_point_start);
+  }
+}
+
+void DrawLineInVertical(ImageView& image, int64_t x_start, int64_t y_start, int64_t line_length,
+                        gsl::span<const uint8_t> color, int64_t thickness) {
+  // boundary check
+  thickness = std::clamp<int64_t>(thickness, 0, image.width - y_start);
+  auto stride = image.width * image.channels;
+  auto point_start = image.data.begin() + x_start * stride + y_start * image.channels;
+  auto point_end = point_start + line_length * stride;
+
+  for (int64_t i = 0; i < thickness; ++i) {
+    std::copy_n(color.begin(), color.size(), point_start + i * image.channels);
+  }
+  for (auto xy_point_start = point_start + stride; xy_point_start < point_end; xy_point_start += stride) {
+    std::copy_n(point_start, thickness * image.channels, xy_point_start);
+  }
+}
+
+// Draw a box on the image with given thickness and color
+void DrawBox(ImageView& image, gsl::span<const float> box, BoundingBoxFormat bbox_mode,
+             gsl::span<const uint8_t> color, int64_t thickness) {
+  // -------(1)------
+  // |              |
+  //(2)            (4)
+  // |              |
+  // -------(3)------
+  float point_x1 = box[0];
+  float point_x2 = box[1];
+  float point_x3 = box[2];
+  float point_x4 = box[3];
+  if (bbox_mode == BoundingBoxFormat::CENTER_XYWH) {
+    point_x1 = box[0] - point_x3 / 2;
+    point_x2 = box[1] - point_x4 / 2;
+    point_x3 = box[0] + point_x3 / 2;
+    point_x4 = box[1] + point_x4 / 2;
+  } else if (bbox_mode == BoundingBoxFormat::XYWH) {
+    point_x1 = box[0];
+    point_x2 = box[1];
+    point_x3 = box[0] + box[2];
+    point_x4 = box[1] + box[3];
+  }
+  int64_t x_start = static_cast<int64_t>(std::round(point_x1));
+  int64_t y_start = static_cast<int64_t>(std::round(point_x2));
+  int64_t x_end = static_cast<int64_t>(std::round(point_x3));
+  int64_t y_end = static_cast<int64_t>(std::round(point_x4));
+
+  auto offset = thickness / 2;
+  x_start -= offset;
+  y_start -= offset;
+  x_end += (thickness - offset);
+  y_end += (thickness - offset);
+
+  x_start = std::max<int64_t>(x_start, 0L);
+  y_start = std::max<int64_t>(y_start, 0L);
+  x_end = std::min<int64_t>(x_end, image.height - 1);
+  y_end = std::min<int64_t>(y_end, image.width - 1);
+
+  thickness = std::clamp<int64_t>(thickness, 1, std::min(x_end - x_start, y_end - y_start));
+  if (x_end - x_start < thickness || y_end - y_start < thickness) {
+    ORTX_CXX_API_THROW("box is too small to draw", ORT_INVALID_ARGUMENT);
+    return;
+  }
+
+  if (x_start + thickness >= image.height || y_start + thickness >= image.width) {
+    ORTX_CXX_API_THROW("box is too small to draw", ORT_INVALID_ARGUMENT);
+    return;
+  }
```

Comment:
Given we clamp the thickness do we need the tests here? If the thickness is == the width or height it's just going to be a solid filled in box isn't it? so we can draw it even though it's not that useful. but this seems like an edge case where the user specifies a really large thickness so drawing a filled in box vs. throwing might be a better outcome.

Reply:
There are two cases which will hit it.
-  Box has zero areas. Say [x1,y1,x2,y2] = [0,0,0,0] or even [1,1,0,0].
- Thickness is too large. So, we are drawing nothing on the image.

Shall we allow it by default?